download-basket/src/main/java/downloadbasket/helpers/SendDownloadDetailsToEmailThread.java

```diff
@@ -92,11 +92,10 @@ public SendDownloadDetailsToEmailThread(OskariLayerService mapLayerService, JSON
 				}
 
 				if (ldz.isDownloadNormalWay()) {
-					ldz.setGetFeatureInfoRequest(OGCServices.getFilter(download, true));
+					ldz.setGetFeatureInfoRequest(OGCServices.getFilter(download, true, mapLayerService));
```

Comment:
You are loading the layer referenced by download.getString("id") in here. You are also passing mapLayerService "down the line" for no other reason than to load the same layer again a few methods later. You should remove the maplayerService parameter and just send the layer you already have available in this method (OskariLayer oskariLayer). This will make the methods cleaner and easier to test.